src/pages/index.js

```diff
@@ -19,32 +21,169 @@ const addCardButton = document.querySelector(".profile__add-button");
 const addCardModal = document.querySelector("#card-edit-modal");
 const addCardForm = addCardModal.querySelector(".modal__form");
 
+const avatarButton = document.querySelector(".profile__avatar-button");
+const avatarModal = document.querySelector("#avatar-edit-modal");
+const avatarForm = avatarModal.querySelector(".modal__form");
+
+const imagePopUp = new PopupWithImage({ popupSelector: "#preview-modal" });
+const popupDelete = new PopupDelete({ popupSelector: "#delete-card-modal" });
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c5770e84-6b43-47e8-8ef9-37352d6891bc",
+    "Content-Type": "application/json",
+  },
+});
+
+const userInfo = new UserInfo({
+  nameSelector: ".profile__name",
+  descriptionSelector: ".profile__description",
+  userAvatar: ".profile__avatar",
+});
+
+let cardSection;
+
+api.getInitialCards().then((initialCards) => {
+  cardSection = new Section(
+    {
+      items: initialCards,
+      renderer: (data) => {
+        const cardElement = new Card(
+          data,
+          "#card-template",
+          handleImageClick,
+          handleDelete,
+          handleLikeCard
+        );
+        cardSection.addItem(cardElement.getCard());
+      },
+    },
+    ".cards__list"
+  );
+  cardSection.renderItems();
+});
+
+api.loadUserInfo().then((userData) => {
+  userInfo.setUserInfo({
+    name: userData.name,
+    about: userData.about,
+  });
+  userInfo.setAvatar({ avatar: userData.avatar });
+});
+
 /* functions */
 const handleImageClick = (data) => {
   imagePopUp.open(data);
 };
 
+function handleDelete(card) {
+  popupDelete.open();
+  popupDelete.setSubmitAction(() => {
+    api
+      .deleteInitialCards(card.id)
+      .then(() => {
+        card.handleDelete();
+        popupDelete.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeCard(card) {
+  if (!card.isLiked) {
+    api
+      .likeCard(card.id)
+      .then(() => {
+        card.isLiked = true;
+        card.updateLikesView();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .unlikeCard(card.id)
+      .then(() => {
+        card.isLiked = false;
+        card.updateLikesView();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
 function handleAddCardFormSubmit(inputValues) {
-  renderCard(inputValues, cardList);
-  addCardPopUp.close();
+  addCardPopUp.renderLoading(true);
+  api
+    .addCard({
+      name: inputValues.name,
+      link: inputValues.link,
+    })
+    .then((data) => {
+      const card = createCard(data);
+      cardSection.addItem(card);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addCardPopUp.renderLoading(false);
+      addCardPopUp.close();
```

Comment:
Needs correction: Сlose modal in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again. 

Fix that for other cases too